shared/src/ledger/eth_bridge/vp/mod.rs

```diff
@@ -27,6 +28,56 @@ where
     pub ctx: Ctx<'ctx, DB, H, CA>,
 }
 
+impl<'ctx, DB, H, CA> EthBridge<'ctx, DB, H, CA>
+where
+    DB: 'static + ledger_storage::DB + for<'iter> ledger_storage::DBIter<'iter>,
+    H: 'static + StorageHasher,
+    CA: 'static + WasmCacheAccess,
+{
+    /// If the bridge's escrow key was changed, we check
+    /// that the balance increased and that the bridge pool
+    /// VP has been triggered. The bridge pool VP will carry
+    /// out the rest of the checks.
+    fn check_escrow(&self, verifiers: &BTreeSet<Address>) -> bool {
+        let escrow_key = balance_key(&xan(), &super::ADDRESS);
+        let escrow_pre: Amount = if let Ok(Some(bytes)) =
+            self.ctx.read_pre(&escrow_key)
+        {
+            BorshDeserialize::try_from_slice(bytes.as_slice())
+                .expect("Deserializing a balance from storage shouldn't fail")
+        } else {
+            tracing::debug!(
+                "Could not retrieve the Ethereum bridge VP's balance from \
+                 storage"
+            );
+            return false;
+        };
+        let escrow_post: Amount = if let Ok(Some(bytes)) =
+            self.ctx.read_pre(&escrow_key)
+        {
+            BorshDeserialize::try_from_slice(bytes.as_slice())
+                .expect("Deserializing a balance from storage shouldn't fail")
```

Comment:
I guess someone could try to write garbage to this key, and it would cause the VP to panic

Reply:
I guess we could propogate this error then.